californium-core/src/main/java/org/eclipse/californium/core/network/EndpointManager.java

```diff
@@ -98,13 +101,15 @@ public static EndpointManager getEndpointManager() {
 	private final Map<String, Endpoint> endpoints = new ConcurrentHashMap<String, Endpoint>();
 
 	/**
-	 * Get default endpoint for provided scheme. If not available, try to create
-	 * one.
+	 * Get default endpoint for provided scheme. 
+	 * 
+	 * If not available, try to create one, if the provide uriScheme is simple "coap".
 	 * 
 	 * @param uriScheme scheme to select endpoint
-	 * @return the default endpoint for the provided scheme, or null, if not
-	 *         available
-	 * @throws IllegalArgumentException if uriScheme is not supported
+	 * @return the default endpoint for the provided scheme
+	 * @throws IllegalArgumentException if uriScheme is not generally not supported
```

Comment:
`IllegalArgumentException` seems to not be thrown at all

Reply:
public synchronized Endpoint getDefaultEndpoint(String uriScheme) {
		// default endpoints
		if (null == uriScheme) {
			uriScheme = CoAP.COAP_URI_SCHEME;
		}
		if (!CoAP.isSupportedScheme(uriScheme)) {
			throw new IllegalArgumentException("URI scheme " + uriScheme + " not supported!");
		}
		uriScheme = uriScheme.toLowerCase();
		Endpoint endpoint = endpoints.get(uriScheme);
		if (null == endpoint) {
			if (CoAP.COAP_SECURE_URI_SCHEME.equalsIgnoreCase(uriScheme)) {
				throw new IllegalStateException("URI scheme " + uriScheme + " requires a previous set connector!");
			} else if (CoAP.COAP_TCP_URI_SCHEME.equalsIgnoreCase(uriScheme)) {
				throw new IllegalStateException("URI scheme " + uriScheme + " requires a previous set connector!");
			} else if (CoAP.COAP_SECURE_TCP_URI_SCHEME.equalsIgnoreCase(uriScheme)) {
				throw new IllegalStateException("URI scheme " + uriScheme + " requires a previous set connector!");
			} else {
				endpoint = new CoapEndpoint();
			}
			try {
				endpoint.start();
				LOGGER.info("created implicit endpoint {} for {}", new Object[] { endpoint.getUri(), uriScheme });
			} catch (IOException e) {
				LOGGER.error("could not create {} endpoint", uriScheme, e);
			}
			endpoints.put(uriScheme, endpoint);
		}
		return endpoint;
	}

If the scheme is not supported at all, you get such a IllegalArgumentException.